sycl/include/sycl/ext/oneapi/experimental/invoke_simd.hpp

```diff
@@ -386,6 +396,14 @@ template <class Callable> constexpr void verify_callable() {
     static_assert(
         !callable_has_ref_arg,
         "invoke_simd does not support callables with reference arguments");
+    constexpr bool callable_has_struct_ret = has_struct_ret(obj);
+    static_assert(
+        !callable_has_struct_ret,
+        "invoke_simd does not support callables returning structures");
+    constexpr bool callable_has_struct_arg = has_struct_arg(obj);
+    static_assert(
+        !callable_has_struct_arg,
+        "invoke_simd does not support callables with structure arguments");
```

Comment:
invoke_simd SPEC allows structures, it is only the problem in GPU RT.
We need some way to enable struct args/rets for experiments, etc without modifying sources. Please add some macro to guard this check.
For example:
```c++
#ifndef __ESIMD_ENABLE_INVOKE_SIMD_STRUCTS
    constexpr bool callable_has_struct_ret = has_struct_ret(obj);
    static_assert(....);

#endif // end !__ESIMD_ENABLE_INVOKE_SIMD_STRUCTS
```

Reply:
just added this in latest commit, we already have `__INVOKE_SIMD_ENABLE_ALL_CALLABLES` so I based the name on that format, thanks

Reply:
Great. Thank you!